src/repository.js

```diff
@@ -187,6 +187,12 @@ module.exports = class Repository {
       .returning(fields)
       .insert(payload)
 
+    //.returning() is not supported by sqlite3 and will not have any effect, so we have to get last inserted row
+    if (this.runner().client && this.runner().client.driverName && (this.runner().client.driverName.includes('sqlite3'))) {
+      const lastInsertedID = await this.knex.raw(`SELECT last_insert_rowid();`)
```

Comment:
Given the async nature of the queries (`.insert(payload)` and `knex.raw('SELECT last_insert_rowid();')` ) it seems it could happen a race condition here: the returned value for `last_insert_rowid()` would be from a different insert, from a different event loop.

Event Loop A: Insert X
Event Loop B: Insert Y
Event Loop A: get `last_insert_rowid` = Y, but should be X

Reply:
You're right.

We should isolate this in some kind of transaction (http://knexjs.org/guide/transactions.html). I`ll open an issue about this.